workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReadersWriters/RequirableBase.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class RequirableBase : IInjectable, IDisposable
+    {
+        private const string AccessedDisposedRequirable =
+            "Tried to access a disposed [Require] SpatialOS MonoBehaviour API object (e.g. Reader, Writer, CommandRequestSender, CommandRequestHandler etc.). " +
+            "Note that [Require] objects are already disposed at the time OnDisable() is called. " +
+            "Also note that user callbacks are deregistered automatically so there is no need to manually deregister them in OnDisable().";
+
+        private readonly ILogDispatcher logDispatcher;
+        private bool isDisposed;
+
+        public RequirableBase(ILogDispatcher logDispatcher)
+        {
+            this.logDispatcher = logDispatcher;
+        }
+
+        protected bool LogErrorIfDisposed()
+        {
+            if (isDisposed)
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent(AccessedDisposedRequirable)
```

Comment:
I can imagine this becoming very spammy if this code path fires. Any ideas for how to mitigate this?

Reply:
Yeah, I can see how this could get spammy. We could save for which [Require] field or MonoBehaviour type we've already logged an error and only log an error once per field/Behaviour but that would mean that we potentially accumulate a bit cache over time. Do you have a suggestion for how we could handle this more elegantly?

Reply:
You can reduce duplicated error spam in the logger itself. Ultimately I think we need to revisit what is an isn't an error though.

Reply:
For the sake of getting this in quickly, are you guys ok with revisiting this later?